### PR TITLE
clusterversion,kvserver,server,security: remove 22.1 gates

### PR DESCRIPTION
**Pull request by celiala**
This commit removes the following 22.1 version gates:

_(✅ denotes explicit LGTM)_

**8/8: initial PR version gates** 
- ✅  kvserver: ProbeRequest, 
- ✅ security,pgwire: SCRAMAuthentication,
- ✅ server: UnsafeLossOfQuorumRecoveryRangeLog,
- ✅ kvserver: AlterSystemProtectedTimestampAddColumn

**8/10 Update: Additionally, these version gates to added to this commit**

- ✅ backupccl: EnableProtectedTimestampsForTenant
- ✅ DeleteCommentsWithDroppedIndexes
- sql/catalog: RemoveIncompatibleDatabasePrivileges
- ✅ kvserver: AddRaftAppliedIndexTermMigration
- ✅ clusterversion: PostAddRaftAppliedIndexTermMigration
- kvserver: DontProposeWriteTimestampForLeaseTransfers
- ~✅ storage: EnablePebbleFormatVersionBlockProperties~ 8/12: TODO in future PR.
- ✅ sql: MVCCIndexBackfiller
- ✅ kvserver: LooselyCoupledRaftLogTruncation
- ✅ sql: EnableDeclarativeSchemaChanger
- ~✅ sql: RowLevelTTL~ 8/12: TODO in future PR.
- ✅ kvserver: EnableNewStoreRebalancer
- ~✅ sql: SuperRegions~ 8/12: TODO in future PR.
- ~changefeedccl: EnableNewChangefeedOptions~ 8/12: TODO in future PR.
- ✅ SpanCountTable
- ✅ spanconfig: PreSeedSpanCountTable, SeedSpanCountTable

The cleanup for these version gates was fairly straight-forward, using the guidance from [21.2 cleanup](https://github.com/cockroachdb/cockroach/pull/74270#issuecomment-1000693734):

> For the most part, if the gates were just simple if !version.IsActive { return x } or something, I just removed the block, and even if it was a little more complicated, like args = [x]; if version { args = append(args, y) }; foo(args) I still tried to mostly inline it such that it looked natural (i.e. remove that append and make it args = [x, y]).

22.1 version gates requiring more nuanced removal will be done in separate PRs.

Partially addresses https://github.com/cockroachdb/cockroach/issues/80663

### CODE REVIEW

**Review by knz — APPROVED** [review]
LGTM for the SCRAMAuthentication changes, with a nit.
Didn't review the other changes.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-:-N91xeB306Hu4oznMd2V:b53m7ay)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @aliher1911, @celiala, @dt, @irfansharif, and @tbg)

___
*[`pkg/sql/pgwire/hba_conf.go` line 364 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N91xXbV6cCO4NinjJ7S:-N91xXbV6cCO4NinjJ7T:bhhokoh) ([raw file](https://github.com/cockroachdb/cockroach/blob/24906bd59e975f607e9ea88cbd4ae5ff0432df45/pkg/sql/pgwire/hba_conf.go#L364)):*
> ```Go
> }
> 
> // chainOptions is an option that combines its argument options.
> ```

Could you leave these two functions in, if only via a `var _ = chainOptions` to prevent the linter warning. I believe @kpatron-cockroachlabs will need them in an upcoming PR.

**Comment by celiala** [comment]
@dt @irfansharif - I'm going to remove the version gates in multiple passes, but let me know if version gate removal + bumping minVersion needs to be done more atomically. Thanks!

**Review by irfansharif — COMMENTED** [review]
They don't need to be done atomically (David and Yahor have been removing other cluster versions in other PRs for ex.)

**Comment by celiala** [comment]
pkg/ccl/serverccl/serverccl_test_/serverccl_test is failing BazelCI. Gonna rerun in case it's a flake.

**Review by celiala — COMMENTED** [review]
TFTR!

Bazel Essential CI is failing for `pkg/ccl/serverccl/serverccl_test_/serverccl_test`. But I see a [history of fails](https://teamcity.cockroachdb.com/test/2384887577415544454?currentProjectId=Cockroach_Ci_Tests&expandTestHistoryChartSection=true) and I don't think this PR impacted the `pkg/ccl/serverccl/` area? 

~Since this is a relatively straight-forward cleanup, and since all other tests pass, I will merge soon to unblock remaining version gate cleanup, if I don't hear from other groups.~

8/10 Update: More version gates were added, which are also relatively straight-forward cleanups. I'm going to try to merge this by EOD Thursday or sometime Friday, in order to bump the minVersion, since we select alpha candidate next week.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-:-N938nSB4JncmWikt_o6:b-qhpksj)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @aliher1911, @dt, @kpatron-cockroachlabs, and @tbg)

___
*[`pkg/sql/pgwire/hba_conf.go` line 364 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N91xXbV6cCO4NinjJ7S:-N938fYH3ZrMPQft2eZw:b-ck6xbz) ([raw file](https://github.com/cockroachdb/cockroach/blob/24906bd59e975f607e9ea88cbd4ae5ff0432df45/pkg/sql/pgwire/hba_conf.go#L364)):*
Done. Added:

> var _ = chainOptions
> var _ = requireClusterVersion

**Comment by celiala** [comment]
bors cancel

File: pkg/kv/kvserver/replica_circuit_breaker.go
Comment (tbg) [inline pkg/kv/kvserver/replica_circuit_breaker.go:99]: If you don'd mind, please remove `br.versionIsActive`, as well as `canEnable()`, since it always returns true.

**Review by tbg — APPROVED** [review]

**Review by HonoreDB — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 6 of 10 files at r2, 1 of 8 files at r5, 1 of 4 files at r6, 3 of 6 files at r7, 33 of 33 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85777)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @aliher1911, @celiala, @dt, @irfansharif, @kpatron-cockroachlabs, @miretskiy, and @rail)

**Comment by celiala** [comment]
I meant to push the additional 8/10 version gate removals to a separate branch, but accidentally pushed to this PR:

Since the 8/10 commits were also straight-forward removals, I will go ahead and keep them in this PR. But this means I'm re-adding @cockroachdb/kv-prs to review the version gates that have been added today.

**Comment by postamar** [comment]
Schema LGTM for:
- clusterversion: remove DeleteCommentsWithDroppedIndexes
- clusterversion,sql: remove MVCCIndexBackfiller
- clusterversion,sql: remove EnableDeclarativeSchemaChanger
- clusterversion,sql: remove RowLevelTTL
- clusterversion,sql: remove SuperRegions

File: pkg/kv/kvserver/stateloader/initial.go
Comment (erikgrinaker) [inline pkg/kv/kvserver/stateloader/initial.go:121]: Would you mind removing this parameter from `WriteInitialReplicaState` and the corresponding `if` condition inside it?

**Review by erikgrinaker — APPROVED** [review]
LGTM for UnsafeLossOfQuorumRecoveryRangeLog, AddRaftAppliedIndexTermMigration, PostAddRaftAppliedIndexTermMigration, EnablePebbleFormatVersionBlockProperties, and LooselyCoupledRaftLogTruncation.

Left a couple of minor requests, but I'm happy to handle those myself too once this merges.

Thanks for doing this!

**Review by irfansharif — APPROVED** [review]
EnableProtectedTimestampsForTenant, SpanCountTable, PreSeedSpanCountTable, SeedSpanCountTable all LGTM.

**Review by celiala — COMMENTED** [review]
> Left a couple of minor requests, but I'm happy to handle those myself too once this merges.

@erikgrinaker - I had some clarification questions on these requests (inline) -- although if tests pass (:crossed_fingers:) with this latest update, I may just merge this :sweat_smile: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-:-N9DsCGp6HAspfKwttAy:bfbdy51)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @aliher1911, @dt, @HonoreDB, @kpatron-cockroachlabs, @miretskiy, @rail, and @tbg)

___
*[`pkg/kv/kvserver/replica_circuit_breaker.go` line 99 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N8zTxb3X70T3vF8tv32-r1-99:-N9DsvOBEKwmGJoLxsoN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/68968087b11b383b1af4be632235bb82036b14b7/pkg/kv/kvserver/replica_circuit_breaker.go#L99)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

If you don'd mind, please remove `br.versionIsActive`, as well as `canEnable()`, since it always returns true.
</blockquote></details>

Done.
___
*[`pkg/kv/kvserver/stateloader/initial.go` line 121 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N9Dp8T11wEVW6QCNUSp:-N9Dp8T11wEVW6QCNUSq:bi2e83e) ([raw file](https://github.com/cockroachdb/cockroach/blob/4ae76f5291388b8df803cd514d270230f209bbcc/pkg/kv/kvserver/stateloader/initial.go#L121)):*
[from @erikgrinaker ]
> Would you mind removing this parameter from `WriteInitialReplicaState` and the corresponding if condition inside it?

👍🏼 can do.

Removing `writeRaftAppliedIndexTerm` as a function parameter mean also removing this preceding conditional in

https://github.com/cockroachdb/cockroach/blob/master/pkg/kv/kvserver/batcheval/cmd_end_transaction.go#L1111-L1117

```
		writeRaftAppliedIndexTerm := false
		if rangeAppliedState.RaftAppliedIndexTerm > 0 {
			writeRaftAppliedIndexTerm = true
		}
```

Confirming that this is also okay?
___
*[`pkg/storage/sst_writer.go` line 70 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N9DoYMW1UBfRQnsIVqI:-N9DoYMW1UBfRQnsIVqJ:b-616m57) ([raw file](https://github.com/cockroachdb/cockroach/blob/4ae76f5291388b8df803cd514d270230f209bbcc/pkg/storage/sst_writer.go#L70)):*
> (Reviewable was unable to map this GitHub line comment to the right spot — sorry!)
> We can update `format :=`  above and drop the default branch here. Or alternatively use `var format ....`

Not sure what the `format` value should be? 
To confirm, I should replace all of this:
```
	format := sstable.TableFormatRocksDBv2
	// Cases are ordered from newer to older versions.
	switch {
	case cs.Version.IsActive(ctx, clusterversion.EnablePebbleFormatVersionRangeKeys):
		format = sstable.TableFormatPebblev2 // Range keys.
	case cs.Version.IsActive(ctx, clusterversion.EnablePebbleFormatVersionBlockProperties):
		format = sstable.TableFormatPebblev1 // Block properties.
	}

```
With this?
```
	format := sstable.TableFormatPebblev2 // Range keys.
```

**Review by erikgrinaker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85777)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @aliher1911, @celiala, @dt, @HonoreDB, @kpatron-cockroachlabs, @miretskiy, @rail, and @tbg)

___
*[`pkg/kv/kvserver/stateloader/initial.go` line 121 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N9Dp8T11wEVW6QCNUSp:-N9Dx_Ew8jWqCAixov0c:bnzuxk4) ([raw file](https://github.com/cockroachdb/cockroach/blob/4ae76f5291388b8df803cd514d270230f209bbcc/pkg/kv/kvserver/stateloader/initial.go#L121)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

[from @erikgrinaker ]
> Would you mind removing this parameter from `WriteInitialReplicaState` and the corresponding if condition inside it?

👍🏼 can do.

Removing `writeRaftAppliedIndexTerm` as a function parameter mean also removing this preceding conditional in

https://github.com/cockroachdb/cockroach/blob/master/pkg/kv/kvserver/batcheval/cmd_end_transaction.go#L1111-L1117

```
		writeRaftAppliedIndexTerm := false
		if rangeAppliedState.RaftAppliedIndexTerm > 0 {
			writeRaftAppliedIndexTerm = true
		}
```

Confirming that this is also okay?

</blockquote></details>

Ah, sorry, I missed that -- let's leave it as it was then.
___
*[`pkg/storage/sst_writer.go` line 70 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N9DoYMW1UBfRQnsIVqI:-N9Dxi8RA1dAluUoMWye:b-mgeork) ([raw file](https://github.com/cockroachdb/cockroach/blob/4ae76f5291388b8df803cd514d270230f209bbcc/pkg/storage/sst_writer.go#L70)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

> (Reviewable was unable to map this GitHub line comment to the right spot — sorry!)
> We can update `format :=`  above and drop the default branch here. Or alternatively use `var format ....`

Not sure what the `format` value should be? 
To confirm, I should replace all of this:
```
	format := sstable.TableFormatRocksDBv2
	// Cases are ordered from newer to older versions.
	switch {
	case cs.Version.IsActive(ctx, clusterversion.EnablePebbleFormatVersionRangeKeys):
		format = sstable.TableFormatPebblev2 // Range keys.
	case cs.Version.IsActive(ctx, clusterversion.EnablePebbleFormatVersionBlockProperties):
		format = sstable.TableFormatPebblev1 // Block properties.
	}

```
With this?
```
	format := sstable.TableFormatPebblev2 // Range keys.
```


</blockquote></details>

Ah, no -- I meant:

```
	format := sstable.TableFormatPebblev1
	// Cases are ordered from newer to older versions.
	switch {
	case cs.Version.IsActive(ctx, clusterversion.EnablePebbleFormatVersionRangeKeys):
		format = sstable.TableFormatPebblev2 // Range keys.
	}
```

Thank you!

**Review by celiala — COMMENTED** [review]
updates:

1) cleaned up switch/case logic for `MakeIngestionWriterOptions1, as per suggested
2) removed failing mixed version tests:
- TestLogic_row_level_ttl_mixed_21_2_22_1
- TestLogic_super_regions_mixed_version
- TestCCLLogic_create_changefeed_mixed_versions

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-:-N9EHYia3Qoy-J3O5U3d:bpj9jh8)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru, @aliher1911, @dt, @erikgrinaker, @HonoreDB, @kpatron-cockroachlabs, @miretskiy, @rail, and @tbg)

___
*[`pkg/kv/kvserver/stateloader/initial.go` line 121 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N9Dp8T11wEVW6QCNUSp:-N9EGvXpCg4YJw_A4vCo:bjmoqms) ([raw file](https://github.com/cockroachdb/cockroach/blob/4ae76f5291388b8df803cd514d270230f209bbcc/pkg/kv/kvserver/stateloader/initial.go#L121)):*
<details><summary><i>Previously, erikgrinaker (Erik Grinaker) wrote…</i></summary><blockquote>

Ah, sorry, I missed that -- let's leave it as it was then.
</blockquote></details>

Ack - leaving `WriteInitialReplicaState` interface as-is :)
___
*[`pkg/storage/sst_writer.go` line 70 at r8](https://reviewable.io/reviews/cockroachdb/cockroach/85777#-N9DoYMW1UBfRQnsIVqI:-N9EHLcC81bcfw2R00KG:bb1mq8j) ([raw file](https://github.com/cockroachdb/cockroach/blob/4ae76f5291388b8df803cd514d270230f209bbcc/pkg/storage/sst_writer.go#L70)):*
<details><summary><i>Previously, erikgrinaker (Erik Grinaker) wrote…</i></summary><blockquote>

Ah, no -- I meant:

```
	format := sstable.TableFormatPebblev1
	// Cases are ordered from newer to older versions.
	switch {
	case cs.Version.IsActive(ctx, clusterversion.EnablePebbleFormatVersionRangeKeys):
		format = sstable.TableFormatPebblev2 // Range keys.
	}
```

Thank you!
</blockquote></details>

Done! (thanks for giving me the code to copy-pasta - super helpful :) !)

File: pkg/kv/kvserver/store_rebalancer.go
Comment (irfansharif) [inline pkg/kv/kvserver/store_rebalancer.go:394]: I believe there are no other usages of deprecatedChooseLeaseToTransfer, @aayushshah15 added it in #80190 to version gate the new 22.1 behaviour by keeping the 21.2 one around, but can be removed in this PR. Likewise for deprecatedChooseRangeToRebalance.

Comment (irfansharif) [reply pkg/kv/kvserver/store_rebalancer.go:394]: The EnableNewStoreRebalancer also LGTM otherwise.

**Comment by celiala** [comment]
TFTRs!

bors r+

8/12 Update: These version gate removals were reverted (removed from this PR) to pass tests. To be addressed separately (future PRs):

- storage: EnablePebbleFormatVersionBlockProperties
- sql: RowLevelTTL
- sql: SuperRegions
- changefeedccl: EnableNewChangefeedOptions

File: pkg/kv/kvserver/store_rebalancer.go
Comment (irfansharif) [reply pkg/kv/kvserver/store_rebalancer.go:394]: Sent a PR: #86060.

**Pull request merged**
